### ISSUE TITLE
rke cluster.yml extra_env values are not updated after changes

### ISSUE THREAD

**Issue by damlub** [rancher/rke#772]
**RKE version:**
rke version v0.1.8 (on Mac)

**Docker version: (`docker version`,`docker info` preferred)**
Client:
 Version:      17.03.2-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 02:21:36 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.2-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   f5ec1e2
 Built:        Tue Jun 27 02:21:36 2017
 OS/Arch:      linux/amd64
 Experimental: false

Containers: 32
 Running: 16
 Paused: 0
 Stopped: 16
Images: 42
Server Version: 17.03.2-ce
Storage Driver: overlay
 Backing Filesystem: extfs
 Supports d_type: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-862.6.3.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.701 GiB
Name: kubernetes01
ID: 6DKQ:2UPH:2X52:OYZU:QJCI:4BH7:7LUC:K3MO:DVXO:A4Z4:X77R:SOCY
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"

3.10.0-862.6.3.el7.x86_64

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
VMs based on KVM

**cluster.yml file:**
(relevant section)
```   ...
   services:
     kube-apiserver:
       extra_args:
         feature-gates: "PersistentLocalVolumes=true,VolumeScheduling=true"
     kubelet:
       extra_args:
         feature-gates: "PersistentLocalVolumes=true,VolumeScheduling=true"
       extra_env:
         - "PROXY=damian1"
```
**Steps to Reproduce:**
Setup a new Rancher Cluster with `rke up`, then change variabe `PROXY=damian2`, rerung `rke up`
The value of the variable can be checked with `docker inspect kubelet|grep -C2 PROXY` on the Linux host.

**Results:**
the kubelets on the nodes do not get restarted, PROXY env var does not change

Changing the extra_args settings is restarting the container and changes the according settings.

**Comment by damlub** [comment]
Any updates on this? Can this be reproduced? Is more information needed? 
This scenario is unfortunate, when you want to change, e.g., the HTTP-Proxy's IP after a cluster has been created. In that case it seems only a resetup will add the according ENV_VARS. Or is there another way to alter the ENV_VARS?

Cheers,
Damian

**Comment by galal-hussein** [comment]
this issue was solved in v0.1.9 using this PR https://github.com/rancher/rke/pull/814

i can confirm that env changed when you change the extra env and re run rke up:
```
docker inspect kubelet | grep FOO
                "FOO=bar",
```
```
docker inspect kubelet | grep FOO
                "FOO=bar2",
```
